components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/OpenIDConnectAuthenticator.java

```diff
@@ -746,8 +747,12 @@ protected OAuthClientResponse requestAccessToken(HttpServletRequest request, Aut
             String accessToken = request.getParameter(ACCESS_TOKEN_PARAM);
             try {
                 validateJWTToken(context, idToken);
-            } catch (ParseException | IdentityOAuth2Exception | JOSEException e) {
-                throw new AuthenticationFailedException("JWT Token validation Failed.");
+            } catch (ParseException | IdentityOAuth2ClientException | JOSEException e) {
+                throw new AuthenticationFailedException(ErrorMessages.INVALID_JWT_TOKEN.getCode(),
+                        ErrorMessages.INVALID_JWT_TOKEN.getMessage());
+            } catch (IdentityOAuth2Exception e) {
+                throw new AuthenticationFailedException(ErrorMessages.JWT_TOKEN_VALIDATION_FAILED.getCode(),
+                        ErrorMessages.JWT_TOKEN_VALIDATION_FAILED.getMessage(), e);
```

Comment:
Seems the message's alias placeholder (%s) is not getting replaced

Reply:
This error message (`JWT_TOKEN_VALIDATION_FAILED`) refers to the string `"JWT token validation Failed."` which does not contain a place holder.